base/steer/FairRootManager.cxx

```diff
@@ -974,70 +978,81 @@ void FairRootManager::DeleteOldWriteoutBufferData()
+
     char* workdir = getenv("VMCWORKDIR");
     if (nullptr == workdir) {
         return default_name;
     }
 
     // Open file with output tree name
-    FILE* file = fopen(Form("%s/config/rootmanager.dat", workdir), "r");
+    std::ifstream file(Form("%s/config/rootmanager.dat", workdir));
     // If file does not exist -> default
-    if (nullptr == file) {
+    if (!file.is_open()) {
         return default_name;
     }
 
-    char str[100];
-    while (nullptr != fgets(str, 100, file)) {
-        if (TString(str).Contains("treename")) {
-            char* treename = new char[100];
-            if (1 == sscanf(str, "treename=%s", treename)) {
-                fclose(file);
-                return treename;
-            }
-            delete[] treename;
+    std::string line;
+    while (std::getline(file, line)) {
+        if (line.rfind(namekind, 0) == 0) {
+            line.replace(0, strlen(namekind), "");
+            file.close();
+            return line;
         }
     }
 
-    // Key "treename" was not found in file: return default
-    fclose(file);
+    file.close();
     return default_name;
 }
 
-char* FairRootManager::GetFolderName()
+char* FairRootManager::GetTreeName()
 {
-    char* default_name = (char*)"cbmout";
-    if (!FairRun::Instance()->IsAna()) {
-        default_name = (char*)"cbmroot";
+    if (fTreeName.length() > 0) {
+        char* treeName = new char[fTreeName.length()];
+        strcpy(treeName, fTreeName.c_str());
+        return treeName;
     }
-    char* workdir = getenv("VMCWORKDIR");
-    if (nullptr == workdir) {
-        return default_name;
+
+    std::string nameFromFile = GetNameFromFile("treename=");
+    if (nameFromFile.length() > 0) {
+        char* treeName = new char[nameFromFile.length()];
+        strcpy(treeName, nameFromFile.c_str());
+        return treeName;
     }
 
-    // Open file with output tree name
-    FILE* file = fopen(Form("%s/config/rootmanager.dat", workdir), "r");
-    // If file does not exist -> default
-    if (nullptr == file) {
-        return default_name;
+    char* default_name = new char[7];
+    strcpy(default_name, "cbmsim");
+    return default_name;
+}
+
+char* FairRootManager::GetFolderName()
+{
+    if (fFolderName.length() > 0) {
+        char* treeName = new char[fFolderName.length()];
+        strcpy(treeName, fFolderName.c_str());
+        return treeName;
     }
 
-    char str[100];
-    while (nullptr != fgets(str, 100, file)) {
-        if (TString(str).Contains("foldername")) {
-            char* foldername = new char[100];
-            if (1 == sscanf(str, "foldername=%s", foldername)) {
-                fclose(file);
-                return foldername;
-            }
-            delete[] foldername;
-        }
+    std::string nameFromFile = GetNameFromFile("foldername=");
+    if (nameFromFile.length() > 0) {
+        char* treeName = new char[nameFromFile.length()];
+        strcpy(treeName, nameFromFile.c_str());
+        return treeName;
```

Comment:
All of that said, I am still not happy with `FairRootManager::Get{Tree,Folder}Name` returning allocated space that probably nobody is going to free (see our own usage a few lines up).

One option would be to use [`Form`](https://root.cern/doc/master/TString_8cxx.html#acb421e430108807d6d57ee3398f84716)'s circular buffer and let the methods return a `const char *`?
```suggestion
        return Form("%s", nameFromFile.c_str());
```
I am quite unsure, really.

Reply:
Looks soo much better!

As the functions now return memory that doesn't need to be freed nor changed: What about returning `const char*`?

Reply:
Thanks!